lib/mpl_toolkits/axisartist/axis_artist.py

```diff
@@ -156,12 +156,12 @@ def draw(self, renderer):
             join = self._solidjoinstyle
         gc.set_joinstyle(join)
         gc.set_capstyle(cap)
+        gc.set_dashes(self._dashOffset, self._dashSeq)
 
-        funcname = self._lineStyles.get(self._linestyle, '_draw_nothing')
```

Comment:
`set_linestyle` (the class here inherits it from lines.Line2D) now has extensive validation before setting `_linestyle`, which means invalid `_linestyle` should never happen.  The same code in `lines` was committed by jdh as far back as 2004, whereas the history of `set_linestyle` is a bit difficult to unwind but I'd guess the checking came later.